src/Controller/CRUDController.php

```diff
@@ -1664,17 +1677,21 @@ final protected function assertObjectExists(Request $request): void
 
         while (null !== $admin) {
             $objectId = $request->get($admin->getIdParameter());
+            if (null === $objectId) {
+                throw $this->createNotFoundException(sprintf(
+                    'Unable to find the %s object id of the admin "%s".',
+                    $admin->getClassnameLabel(),
+                    \get_class($admin)
+                ));
+            }
 
-            if (null !== $objectId) {
-                $adminObject = $admin->getObject($objectId);
-
-                if (null === $adminObject) {
-                    throw $this->createNotFoundException(sprintf(
-                        'Unable to find %s object with id: %s.',
-                        $admin->getClassnameLabel(),
-                        $objectId
-                    ));
-                }
+            $adminObject = $admin->getObject($objectId);
+            if (null === $adminObject) {
```

Comment:
why do you need an extra assert if you are checking here? Maybe make this method final fixes it?

Reply:
The method is already final.
But phpstan/psalm doesn't know what's happening in the method.

Reply:
Plus (next commit), the check cannot be done for list and create actions.